### PR TITLE
Report memory usage by application

### LINKED ISSUE #7866: Breakdown usage by managed application
Currently the usage data reporting is summary only https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-licensing.html#k8s-get-usage-data

We should create a breakdown by managed application, as different license agreements per user can influence whether or not e.g. Logstash memory should be included in the ERU summary. The intent behind this change is to make it transparent to users which applications are being included in the calculatation and allow users that are on older license terms to subtract e.g. Logstash from the total displayed.

### PR DESCRIPTION
**Pull request by pebrc**
Fixes #7866 

Changes the license reporting to not only show the total managed memory but also show a breakdown by application.

### CODE REVIEW

File: pkg/license/license.go [hunk 1]
```diff
@@ -37,29 +37,76 @@ const (
...
+	appUsage    map[string]managedMemory
+	totalMemory managedMemory
+}
+
+func newMemoryUsage() memoryUsage {
+	usage := memoryUsage{
+		appUsage:    map[string]managedMemory{},
+		totalMemory: managedMemory{label: totalKey},
+	}
+	return usage
```
Comment (barkbay) [inline pkg/license/license.go:80]: nit: usage variable is not used.

```suggestion
	return memoryUsage{
		appUsage:    map[string]managedMemory{},
		totalMemory: managedMemory{label: totalKey},
	}
```

File: pkg/utils/metrics/operator.go [hunk 2]
```diff
@@ -52,6 +52,46 @@ var (
...
+
+	// LicensingAPMMemoryGauge reports the APM server memory usage for licensing purposes.
+	LicensingAPMMemoryGauge = registerGauge(prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: namespace,
+		Subsystem: licensingSubsystem,
+		Name:      "memory_gibibytes_apm",
+		Help:      "Memory used by APM server in GiB",
+	}, []string{LicenseLevelLabel}))
+
+	// LicensingEntSearchMemoryGauge reports the Enterprise search memory usage for licensing purposes.
```
Comment (barkbay) [inline pkg/utils/metrics/operator.go:80]:
```suggestion
		Name:      "memory_gibibytes_apm",
		Help:      "Memory used by APM server in GiB",
	}, []string{LicenseLevelLabel}))

	// LicensingEntSearchMemoryGauge reports the Enterprise Search memory usage for licensing purposes.
```

**Review by barkbay — APPROVED** [review]
LGTM

**Comment by pebrc** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.14|[](https://github.com/elastic/cloud-on-k8s/pull/7967)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**